subsys/bluetooth/host/id.c

```diff
@@ -775,7 +775,10 @@ void bt_id_add(struct bt_keys *keys)
 
 	err = hci_id_add(keys->id, &keys->addr, keys->irk.val);
 	if (err) {
-		BT_ERR("Failed to add IRK to controller");
+		/* Duplicate device is OK */
+		if (err != BT_HCI_ERR_INVALID_PARAM) {
```

Comment:
The fix does not work. The `hci_id_add` returns Zephyr error code (`-EIO` == `-5`) so the comparison would not catch it.
The actual HCI error code is not returned by the `hci_id_add`, so maybe we could handle the error in the `hci_id_add`?

Also is the `BT_HCI_ERR_INVALID_PARAM` returned only on duplicated IRK? If not then we may hide other errors from the user which is not good.

Reply:
Cheers for the input.  I'll look at resolving the problem another way.